x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/__mocks__/test_adapters.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { responseMock } from './response_factory';
+
+type ResponseMock = ReturnType<typeof responseMock.create>;
+type Method = keyof ResponseMock;
+
+type MockCall = any; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+interface ResponseCall {
+  body: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  status: number;
+}
+
+interface Response extends ResponseCall {
+  calls: ResponseCall[];
+}
+
+const buildResponses = (method: Method, calls: MockCall[]): ResponseCall[] => {
+  if (!calls.length) return [];
+
+  switch (method) {
+    case 'ok':
+      return calls.map(([call]) => ({ status: 200, body: call?.body }));
+    case 'custom':
+      return calls.map(([call]) => ({
+        status: call?.statusCode,
+        body: JSON.parse(call?.body),
```

Comment:
I know that these question marks never evaluate to `undefined/void` because if you do:

```ts
JSON.parse(undefined),
```

Throws an exception.

I changed it to remove the extra `?` and it looked cleaner and all the tests still evaluate correctly:

```ts
const buildResponses = (method: Method, calls: MockCall[]): ResponseCall[] => {
  if (!calls.length) return [];

  switch (method) {
    case 'ok':
      return calls.map(([call]) => ({ status: 200, body: call.body }));
    case 'custom':
      return calls.map(([call]) => ({
        status: call.statusCode,
        body: JSON.parse(call.body),
      }));
    default:
      throw new Error(`Encountered unexpected call to response.${method}`);
  }
};
```

Reply:
Good catch, this was a relic from when I was capturing calls that legitimately had no body.

Reply:
65eec4acd5f46591917ad33320954b8513c3090f